patches/tModLoader/Terraria/ModLoader/DamageClass.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace Terraria.ModLoader
+{
+	public abstract class DamageClass : ModType
+	{
+		internal int index;
+
+		/// <summary>
+		/// This is the name that will show up when an item tooltip displays 'X [name] damage'.
+		/// </summary>
+		public virtual string ClassName => Name;
+
+		protected override void Register() {
+			Mod.damageClasses[Name] = this;
+			index = DamageClassLoader.Add(this);
+			ContentInstance.Register(this);
```

Comment:
Merge latest 1.4 branch, remove Register (automatic now) and add `ModTypeLookup<DamageClass>.Register`